lua/lazyvim/plugins/extras/lang/fsharp.lua

```diff
@@ -0,0 +1,63 @@
+-- requires dotnet sdk  to be installed
+return {
+  recommended = function()
+    return LazyVim.extras.wants({
+      ft = { "fs", "fsx", "fsi" },
+      root = { "*.fsproj" },
+    })
+  end,
+  {
+    "nvim-treesitter/nvim-treesitter",
+    config = function()
+      local parser_config = require("nvim-treesitter.parsers").get_parser_configs()
+      parser_config.fsharp = {
+        install_info = {
+          url = "https://github.com/ionide/tree-sitter-fsharp",
+          branch = "main",
+          files = { "src/scanner.c", "src/parser.c" },
+        },
+        filetype = "fsharp",
+      }
+    end,
+    opts = { ensure_installed = { "fsharp" } },
```

Comment:
Would be best to have this in nvim-treesitter first

Reply:
It is a custom parser defined above

Reply:
adding the parser is useless without any queries, this should be PRed first to nvim-treesitter if the parser is of good enough quality and can parse most of the F# spec, then can be added here

Reply:
> adding the parser is useless without any queries, this should be PRed first to nvim-treesitter if the parser is of good enough quality and can parse most of the F# spec, then can be added here

I didn't add any query, this plugin has queries in it's repo

Reply:
queries in its repo =/= queries that neovim supports and/or are in nvim-treesitter

Reply:
Is there a way to add queries with extras?

Reply:
Not directly, and you shouldn't. Again, it should be added to nvim-treesitter first if the parser is of good enough quality.

Reply:
Dude this thing works and that's what ppl care about who are new to neovim and LazyVim, idk if that parser is good or not because I don't do fsharp but have done csharp so I have a general idea of dotnet ecosystem,  there isn't any other person working on fsharp parser so there isn't a choice

Reply:
> Dude this thing works

It does *not* work, and if you don't do f-sharp then maybe don't add the parser without bothering to understand how tree-sitter/nvim-treesitter work? Adding it here does *absolutely* nothing without proper queries